testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -15906,6 +15909,9 @@ void testTimestampDiff(boolean coercionEnabled) {
     f.checkType("variance(cast(null as varchar(2)))", "DECIMAL(19, 9)");
     f.checkType("variance(CAST(NULL AS INTEGER))", "INTEGER");
     f.checkAggType("variance(DISTINCT 1.5)", "DECIMAL(2, 1) NOT NULL");
+
+    final String[] values2 = {"cast(64.34 as double)", "64.34", "64.34"};
+    f.checkAgg("variance(x)", values2, isExactly(0));
```

Comment:
This looks somewhat dangerous; FP arithmetic is rarely exact.
For example, I have seen differences between x86 and AARCH64.
But if it works now, probably it will continue to work.